### PR TITLE
Fix GCS Keystore Handling in FIPS Mode

### LINKED ISSUE #75023: [CI] GoogleCloudStorageThirdPartyTests testCleanup failing
This is the result of a changed exception type after upgrading the GCS dependency I think. Looking into it now.

**Build scan:**
https://gradle-enterprise.elastic.co/s/venckon4avoas/tests/:plugins:repository-gcs:gcsThirdPartyTest/org.elasticsearch.repositories.gcs.GoogleCloudStorageThirdPartyTests/testCleanup

**Reproduction line:**
`./gradlew ':plugins:repository-gcs:gcsThirdPartyTest' --tests "org.elasticsearch.repositories.gcs.GoogleCloudStorageThirdPartyTests.testCleanup" -Dtests.seed=6316403C7D97DB97 -Dtests.locale=mk -Dtests.timezone=Africa/Bangui -Druntime.java=11 -Dtests.fips.enabled=true`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.repositories.gcs.GoogleCloudStorageThirdPartyTests&tests.test=testCleanup

**Failure excerpt:**
```
org.elasticsearch.repositories.RepositoryVerificationException: [test-repo] path [integration_test_third_party_tests_6316403C7D97DB97] is not accessible on master node


  Caused by: java.lang.ClassCastException: class java.security.KeyStoreException cannot be cast to class java.io.IOException (java.security.KeyStoreException and java.io.IOException are in module java.base of loader 'bootstrap')

    at org.elasticsearch.repositories.gcs.SocketAccess.doPrivilegedIOException(SocketAccess.java:35)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageService.createClient(GoogleCloudStorageService.java:125)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageService.client(GoogleCloudStorageService.java:105)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStore.client(GoogleCloudStorageBlobStore.java:117)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStore.lambda$writeBlobMultipart$8(GoogleCloudStorageBlobStore.java:455)
    at org.elasticsearch.repositories.gcs.SocketAccess.lambda$doPrivilegedVoidIOException$0(SocketAccess.java:43)
    at java.security.AccessController.doPrivileged(AccessController.java:-2)
    at org.elasticsearch.repositories.gcs.SocketAccess.doPrivilegedVoidIOException(SocketAccess.java:42)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStore.writeBlobMultipart(GoogleCloudStorageBlobStore.java:454)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStore.writeBlob(GoogleCloudStorageBlobStore.java:244)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainer.writeBlob(GoogleCloudStorageBlobContainer.java:78)
    at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainer.writeBlobAtomic(GoogleCloudStorageBlobContainer.java:91)
    at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1617)
    at org.elasticsearch.repositories.RepositoriesService$4.doRun(RepositoriesService.java:393)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:737)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:834)

```

### PR DESCRIPTION
**Pull request by original-brownbear**
Fallout from upgrading the GCS dep. in #74938

In FIPS mode loading the `.p12` keystore used by the new SDK version is not supported
because of "PBE AlgorithmParameters not available". Fortunately, the SDK still includes
the old jks trust store so we can just manually load it the same way it was loaded by
the previous version to fix things.
Also, fixed `SocketAccess` to properly rethrow this kind of exception and not run into
a class cast issue.

Closes #75023

Relates googleapis/google-api-java-client#1738

non-issue as this hasn't been released yet

### CODE REVIEW

File: plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java [hunk 1]
```diff
@@ -126,7 +128,11 @@ private Storage createClient(GoogleCloudStorageClientSettings clientSettings,
             final NetHttpTransport.Builder builder = new NetHttpTransport.Builder();
             // requires java.lang.RuntimePermission "setFactory"
             // Pin the TLS trust certificates.
-            builder.trustCertificates(GoogleUtils.getCertificateTrustStore());
+            final KeyStore certTrustStore = SecurityUtils.getJavaKeyStore();
+            try (InputStream keyStoreStream = GoogleUtils.class.getResourceAsStream("google.jks")) {
```
Comment (fcofdez) [inline plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java:132]: nit: Could you add a comment explaining why we have to load this key store?

**Review by fcofdez — APPROVED** [review]
LGTM 👍

File: plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java (hunk 1 above)
Comment (tlrx) [inline plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java:132]: Can we add a small comment why `GoogleUtils.getCertificateTrustStore()` is not used directly?

**Review by tlrx — APPROVED** [review]
LGTM - left a minor comment

**Comment by original-brownbear** [comment]
Thanks Francisco + Tanguy!

**Pull request merged**